pints/_mcmc/_slice_stepout.py

```diff
@@ -0,0 +1,430 @@
+        # limit of expansion steps
+
+        # Check whether we can expand to the left
+        if self._j > 0 and self._current_log_y < self._fx_l:
+
+            # Set flag to indicate that we are updating the left edge
+            self._set_l = True
+
+            # If left edge of the interval is inside the slice, keep expanding
+            self._l -= self._w[self._i]
+            self._temp_l[self._i] = self._l
+            self._j -= 1
+
+            # Ask for log pdf of the updated left edge
+            self._ready_for_tell = True
+
+            return np.array(self._temp_l, copy=True)
+
+        # Reset flag now that we have finished updating the left edge
+        self._set_l = False
+
+        # Check whether we can expand to the right
+        if self._k > 0 and self._current_log_y < self._fx_r:
+
+            # Set flag to indicate that we are updating the right edge
+            self._set_r = True
+
+            # If right edge of the interval is inside the slice, keep expanding
+            self._r += self._w[self._i]
+            self._temp_r[self._i] = self._r
+            self._k -= 1
+
+            # Ask for log pdf of the updated right edge
+            self._ready_for_tell = True
+
+            return np.array(self._temp_r, copy=True)
+
+        # Reset flag now that we have finished updating the right edge
+        self._set_r = False
+
+        # Now that we have expanded the interval, set flag
+        self._interval_found = True
+
+        # Sample new trial point by sampling uniformly from the interval ``I=(l,r)``
+        self._u = np.random.uniform()
+        self._proposed[self._i] = self._l + self._u*(self._r - self._l)
+
+        # Send trial point for checks
+        self._ready_for_tell = True
+        return np.array(self._proposed, copy=True)
+
+
+    def tell(self, reply):
+        """ See :meth:`pints.SingleChainMCMC.tell()`. """
+
+        # Check ask/tell pattern
+        if not self._ready_for_tell:
+            raise RuntimeError('Tell called before proposal was set.')
+        self._ready_for_tell = False
+
+        # Unpack reply
+        fx = np.asarray(reply, dtype=float)
+
+        # Very first call
+        if self._current is None:
+
+            # Check first point is somewhere sensible
+            if not np.isfinite(fx):
+                raise ValueError(
+                    'Initial point for MCMC must have finite logpdf.')
+
+            # Set current sample, log pdf of current sample and initialise proposed
+            # sample for next iteration
+            self._current = np.array(self._x0, copy=True)
+            self._current_log_pdf = fx
+            self._proposed = np.array(self._current, copy=True)
+
+            # Sample height of the slice log_y for next iteration
+            self._e = np.random.exponential(1)
+            self._current_log_y = self._current_log_pdf - self._e
+
+            # Increase number of samples found
```

Comment:
Isn't this handled in a more abstract method for MCMC samplers?

Reply:
Yes, I checked and I believe that it is handled in the run() method of the _mcmc/__init__.py script. I have removed every line concerning updating the mcmc iteration count in my code.